### PR TITLE
Feature: share modal

### PR DESCRIPTION
**Pull request by edbrett**
- share modal component added with reducer and actions and container
- open modal component and container
- responsive styles
- bread crumb adjustments

### CODE REVIEW

File: app/assets/javascripts/components/pages/CampaignsPageView.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import React from 'react';
 import DataTableView from './../DataTableView'
 
-class CampaingView extends React.Component {
+class CampaignView extends React.Component {
```
Comment (simaob) [inline app/assets/javascripts/components/pages/CampaignsPageView.js:6]: Well done on the typo! =D Damn Campaign the worst word....

File: app/assets/javascripts/components/OpenModal.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, {Component} from 'react';
+
+class OpenModal extends Component {
+  render() {
+    console.log(this.props);
```
Comment (simaob) [inline app/assets/javascripts/components/OpenModal.js]: let's tidy up before merging.

**Comment by simaob** [comment]
I think it would be a great addition if we could close the modal by hitting the _ESC_ on the keyboard. Could you add that, please?

**Comment by edbrett** [comment]
Agreed! All done for you :)

**Pull request merged**